### PR TITLE
Allow user to undo any current edit session

### PR DESCRIPTION
**Pull request by jshum2479**
Added parameter -undo_current_edit for online update and deploy.  If the parameter is present, any outstanding unactivated changes will be discarded before proceeding.

### CODE REVIEW

File: core/src/main/python/update.py [hunk 1]
```diff
@@ -125,9 +126,11 @@ def __update_online(model, model_context, aliases):
 
     try:
         __wlst_helper.connect(admin_user, admin_pwd, admin_url)
-        deployer_utils.ensure_no_uncommitted_changes_or_edit_sessions()
+        deployer_utils.ensure_no_uncommitted_changes_or_edit_sessions(model_context.is_undo_current_edit())
         __wlst_helper.edit()
         __wlst_helper.start_edit()
+        if model_context.is_undo_current_edit():
+            deployer_utils.undo_current_edit()
```
Comment (rakillen) [inline core/src/main/python/update.py:133]: Does undo_current_edit need to happen before the new edit is started?

File: core/src/main/python/deploy.py [hunk 2]
```diff
@@ -121,7 +122,7 @@ def __deploy_online(model, model_context, aliases):
 
     try:
         __wlst_helper.connect(admin_user, admin_pwd, admin_url)
-        deployer_utils.ensure_no_uncommitted_changes_or_edit_sessions()
+        deployer_utils.ensure_no_uncommitted_changes_or_edit_sessions(model_context.is_undo_current_edit())
```
Comment (rakillen) [inline core/src/main/python/deploy.py:125]: Deploy should do check and do deployer_utils.undo_current_edit same as update.

File: core/src/main/python/update.py (hunk 1 above)
Comment (jshum2479) [reply core/src/main/python/update.py:133]: Yes, it needs to have an edit session first

File: core/src/main/python/deploy.py (hunk 2 above)
Comment (jshum2479) [reply core/src/main/python/deploy.py:125]: added

**Review by rakillen — APPROVED** [review]

**Review by CarolynRountree — APPROVED** [review]

**Review by ddsharpe — APPROVED** [review]

**Pull request merged**